types/react-onclickoutside/index.d.ts

```diff
@@ -11,8 +11,8 @@ export interface HandleClickOutside<T> {
 }
 
 export interface InjectedOnClickOutProps {
-    disableOnClickOutside(): void;
-    enableOnClickOutside(): void;
+    disableOnClickOutside?(): void;
+    enableOnClickOutside?(): void;
```

Comment:
These properties are not optional but are injected by component inside HoC and will always be present: https://github.com/Pomax/react-onclickoutside/blob/master/src/index.js#L201-L202

Reply:
Can you explain why I and several others are getting the error shown here, then? https://github.com/Pomax/react-onclickoutside/issues/171

If I don't have my component extend `InjectedOnClickOutProps` then I get the error in the first post on that link, but if I do I get

```
ERROR in [at-loader] ./source/components/search-react/index.tsx:184:35 
    TS2322: Type '{ allBrands: Brand[]; selectedBrands: () => Brand[]; isSelected: (brand: Brand) => boolean; hideB...' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<BrandsDropdownProps & OnClickOutProps, C...'.
  Type '{ allBrands: Brand[]; selectedBrands: () => Brand[]; isSelected: (brand: Brand) => boolean; hideB...' is not assignable to type 'Readonly<BrandsDropdownProps & OnClickOutProps>'.
    Property 'disableOnClickOutside' is missing in type '{ allBrands: Brand[]; selectedBrands: () => Brand[]; isSelected: (brand: Brand) => boolean; hideB...'.
```

So it doesn't appear to be automatically being injected.

Reply:
It's probably the way how oyu define your inner component in Typescript. Seems like your props type don't match and they are expected to extend `InjectedOnClickProps`. This is the example of how it should loook more or less (it's a little verbose to get the whole picture):

```typescript
import * as React form "react";
import onClickOutside form "react-onclickoutside";

type InjectedProps = InjectedOnClickProps;

interface OwnProps {
  foo?: string;
}

type MyComponentProps = OwnProps & InjectedProps;

// stateless component case
const MyComponentStateless = onClickOutside<OwnProps>(({  
    foo,
    enableOnClickOutside,
    disableOnClickOutside
}) => {
   disableClickOutside();
   return <div>{ foo }</div>;
});

// class component case
const MyComponentClass = onClickOutside(
    class extends React.Component<MyComponentProps> {
        componentWillMount() {
            this.props.disableClickOutside();
        }

        render() {
            return <div>{ this.props.foo }</div>;
        }
    }
)
```

In that case the types match and you can access injected props inside your component, yet outer component only accepts outer props which will equals to `OwnProps`.

Reply:
Thank you, I was expecting this to work: 
```
interface OwnProps extends InjectedOnClickProps {
    ...
}

class MyComponent extends React.Component<OwnProps, {}> { 
    ...
    constructor (props: OwnProps) {
    ...
```

Switching to using the `&` operator had it working.

```
interface OwnProps {...}

class MyComponent extends React.Component<OwnProps & InjectedOnClickProps, {}> {
    ...
    constructor (props: OwnProps & InjectedOnClickProps) {
    ...
```

Will close this.

Reply:
@LKay do you have any insight on why extending `InjectedOnClickProps` doesn't work as described above? I'm still confused and haven't been able to figure out why an intersection type behaves differently from the extended interface.